rts/motoko-rts/src/region.rs

```diff
@@ -948,6 +948,32 @@ pub(crate) unsafe fn region_load<M: Memory>(_mem: &mut M, r: Value, offset: u64,
     }
 }
 
+pub unsafe fn region_write_heap_shapshot<M: Memory>(mem: &mut M, r: Value) {
+    use crate::constants::WASM_PAGE_SIZE;
+    use core::arch::wasm32;
+
+    let heap_size: usize = wasm32::memory_size(0) * (WASM_PAGE_SIZE.as_u32() as usize);
```

Comment:
The variable could maybe be named `memory_size` as it covers the entire allocated memory (including unused space beyond the heap end). Perhaps, the function itself could be named `memory_snapshot` or `main_memory_snapshot`. Alternatively, we could also only focus on the heap (static and dynamic heap).